terminus-ui/chip/src/chip-collection.component.spec.ts

```diff
@@ -132,44 +134,73 @@ describe(`TsChipCollection`, function() {
     expect(instance.value).toEqual(['banana']);
   });
 
+  describe(`empty`, () => {
+
+    test(`should check the number of chips if the collection exists`, () => {
+      fixture = createComponent(testComponents.NoChip);
+      chipCollectionInstance = getChipCollectionInstance(fixture);
+      fixture.detectChanges();
+
+      expect(chipCollectionInstance.empty).toEqual(true);
+    });
+
+  });
+
   test(`should be able to set isSelectable from chip collection and populate to chips`, fakeAsync(function() {
-    setupStandardCollection('RegularChip');
+    setupStandardCollection(testComponents.RegularChip);
 
     chipCollectionInstance.isSelectable = true;
     const chip = chips.toArray()[0];
     fixture.detectChanges();
+
     chipInstance.select();
     tick(1000);
     fixture.detectChanges();
 
     expect(chip.chipCollectionSelectable).toBeTruthy();
   }));
 
-  test(`should emit remove event when chip is removed`, fakeAsync(function() {
-    setupStandardCollection('RegularChip');
-    const chip = fixture.debugElement.query(By.css('.ts-chip')).nativeElement;
-    const chipRemovalButton = chip.querySelector('.ts-chip-remove');
-    chipCollectionInstance.remove = jest.fn();
-    chipCollectionInstance.removed.subscribe(event => {
-      expect(event).toEqual('apple');
-    });
+  test(`should emit 'collectionChange' and 'removed' events when a chip is removed`, function() {
+    setupStandardCollection(testComponents.Events);
+    const buttons = fixture.nativeElement.querySelectorAll('.c-chip__remove');
+    const chipRemovalButton = buttons[1];
+    fixture.componentInstance.change = jest.fn();
+    fixture.componentInstance.removed = jest.fn();
 
-    dispatchMouseEvent(chipRemovalButton, 'click');
-    tick(1000);
+    expect(chipCollectionInstance.chips.length).toEqual(2);
+
+    chipRemovalButton.click();
+    fixture.detectChanges();
+
+    expect(fixture.componentInstance.change.mock.calls[0][0].source.chips.length).toEqual(1);
+    expect(fixture.componentInstance.removed.mock.calls[0][0].chip.value).toEqual('banana');
```

Comment:
What does `.mock.calls` return?

Reply:
An array of the calls made to the spy https://jestjs.io/docs/en/mock-functions#using-a-mock-function